src/main/java/duke/tasks/DeadlineTask.java

```diff
@@ -0,0 +1,38 @@
+package duke.tasks;
+
+import duke.commands.Commands;
+import duke.exceptions.InvalidParameterException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class DeadlineTask extends Task {
+    private final LocalDate deadline;
+
+    private DeadlineTask(String taskName, LocalDate deadline) {
+        super(taskName, "D");
+        this.deadline = deadline;
+    }
+
+    public static DeadlineTask DeadlineTaskBuilder(String taskName) throws InvalidParameterException {
```

Comment:
I understand the rationale behind the naming, but again, module convention states that method names should be presented as a verb.
```suggestion
    public static DeadlineTask buildDeadLineTask(String taskName) throws InvalidParameterException {
```